### PR TITLE
Move author identifiers to version-controlled YAML file

### LINKED ISSUE #9666: Move config/author.yml out of infogami, into version controlled file
### A clear and concise description of what you want to happen

This means moving the author identifiers out of infogami and into a separate file in our repo, similar to what we did with editions in #9234 and #9483

The process should be near identical to the solution we implemented in #9483 for editions.yml.

### Stakeholders

<hr>

### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

### PR DESCRIPTION
**Pull request by fakegithub01**
Closes #9666 

Refactor

### Technical

- Created a new YAML file: `openlibrary/plugins/openlibrary/config/author/author.yml`.
- Updated `_get_author_config()` function to load author identifiers from the new `author.yml` file.

### Testing

- Verify that the application correctly loads author configurations from `author.yml`.
- Test relevant parts of the application to ensure no disruption of existing functionality.

### Stakeholders
@jimchamp @mekarpeles

### CODE REVIEW

File: openlibrary/plugins/openlibrary/config/author/author.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+created:
+    type: /type/datetime
+    value: '2021-05-06T22:11:18.795025'
```
Comment (jimchamp) [inline openlibrary/plugins/openlibrary/config/author/author.yml:3]:
```suggestion
```
We only want the `identifiers` in this file, as per the requirements.

File: openlibrary/plugins/openlibrary/config/author/author.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    url: https://www.youtube.com/@@@
+    website: https://www.youtube.com
+key: /config/author
+last_modified:
+    type: /type/datetime
+    value: '2024-06-17T20:50:26.249752'
+latest_revision: 14
+revision: 14
+type:
+    key: /type/object
```
Comment (jimchamp) [inline openlibrary/plugins/openlibrary/config/author/author.yml:73]:
```suggestion

```
Remove unnecessary fields.  Ensure that the file ends with a newline character.

File: openlibrary/plugins/upstream/utils.py [hunk 3]
```diff
@@ -1222,13 +1222,14 @@ def _get_author_config():
...
     Any changes to /config/author page require restarting the app.
-
     """
-    thing = web.ctx.site.get('/config/author')
-    if hasattr(thing, "identifiers"):
-        identifiers = [Storage(t.dict()) for t in thing.identifiers if 'name' in t]
-    else:
-        identifiers = {}
+    # Load the author config from the author.yml file in the author directory
+    with open('openlibrary/plugins/openlibrary/config/author/author.yml') as in_file:
```
Comment (jimchamp) [inline openlibrary/plugins/upstream/utils.py:1227]:
```suggestion
    with open('openlibrary/plugins/openlibrary/config/author/identifiers.yml') as in_file:
```
Update name of author identifier configuration file to follow same pattern as the edition identifier config file.

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks @fakegithub01!  Will QA this branch after the requested changes have been made.

**Review by Freso — CHANGES_REQUESTED** [review]
You didn’t actually rename the `author.yml` file as you changed the `open()` call.

**Comment by fakegithub01** [comment]
@Freso @jimchamp I have renamed the `author.yml` file to `identifiers.yml` as reviewed for changes. Can you please review the PR?

File: openlibrary/plugins/upstream/utils.py [hunk 4]
```diff
@@ -1222,13 +1222,16 @@ def _get_author_config():
...
     The results are cached on the first invocation.
     Any changes to /config/author page require restarting the app.
-
     """
-    thing = web.ctx.site.get('/config/author')
-    if hasattr(thing, "identifiers"):
-        identifiers = [Storage(t.dict()) for t in thing.identifiers if 'name' in t]
-    else:
-        identifiers = {}
+    # Load the author config from the author.yml file in the author directory
```
Comment (Freso) [inline openlibrary/plugins/upstream/utils.py:1226]: We’re only loading `identifiers` of the author config here… but that’s also currently the entirety of the config. I don’t know if that’s something that needs changing or not.

File: openlibrary/plugins/upstream/utils.py [hunk 5]
```diff
@@ -1222,13 +1222,16 @@ def _get_author_config():
 
     The results are cached on the first invocation.
     Any changes to /config/author page require restarting the app.
```
Comment (Freso) [inline openlibrary/plugins/upstream/utils.py:1224]: The `/config/author` page isn’t getting loaded anymore, so restarting won’t do much… :)

**Review by Freso — COMMENTED** [review]
Seems fine to me. I have left a couple of comments about related documentation that is _slightly_ off now, but I don’t know if Jim will want that changed or not. I just wanted to note the filename so once Jim is available to review this, he wouldn’t have to point that out, so you could get this PR done with sooner. :)

**Review by jimchamp — APPROVED** [review]
Thanks, @fakegithub01!
I've tested this locally by adding a new author identifier entry.  Verified that the new ID is present in the identifiers drop-down in the author edit screen.  Newly added IDs were also present on the author view page.

Will open a follow-up PR that updates the documentation for these functions, and the equivalent edition identifier configuration functions, which, I suspect, were not updated.

**Pull request merged**